agent/xds/rbac.go

```diff
@@ -256,6 +282,11 @@ func intentionToIntermediateRBACForm(
 
 type intentionAction int
 
+type JWTInfo struct {
+	Claims     []*structs.IntentionJWTClaimVerification
+	PayloadKey string
```

Comment:
```suggestion
	MetadataKey string
```

Or maybe `PayloadMetadataKey`. `MetadataPayloadKey`?